### PR TITLE
NAS-130507 / 24.10 / Remove legacy pool.dataset.permission endpoint

### PR DESCRIPTION
**Pull request by anodos325**
This functionality has been superseded by filesystem ACL and permissions endpoints since TrueNAS 12. Endpoint is no longer used by the webui and so it is safe to remove.

### CODE REVIEW

**Review by yocalebo — APPROVED** [review]

**Review by mgrimesix — APPROVED** [review]

File: src/middlewared/middlewared/plugins/pool_/dataset_quota_and_perms.py [hunk 1]
```diff
@@ -10,181 +10,6 @@ class PoolDatasetService(Service):
     class Config:
         namespace = 'pool.dataset'
 
-    @accepts(
-        Str('id', required=True),
-        Dict(
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/pool_/dataset_quota_and_perms.py:15]: FWIW, removing all this code means that now this file has many imported but not used flake8 warnings.
- `errno`
- `Bool`
- `OROperator`
- `UnixPerm`
- `CallError`

File: tests/api2/test_347_posix_mode.py [hunk 2]
```diff
@@ -43,202 +41,127 @@
...
-def test_02_create_dataset(request):
-    result = POST(
-        '/pool/dataset/', {
-            'name': MODE_DATASET
-        }
-    )
-    assert result.status_code == 200, result.text
+def get_mode_octal(path):
+    mode = call('filesystem.stat', path)['mode']
+    server_mode = f"{stat.S_IMODE(mode):03o}"
```
Comment (bmeagherix) [inline tests/api2/test_347_posix_mode.py:82]: `get_mode_octal` is duplicated (sort of) below.

File: tests/api2/test_347_posix_mode.py [hunk 3]
```diff
@@ -43,202 +41,122 @@
...
```
Comment (bmeagherix) [inline tests/api2/test_347_posix_mode.py:82]: Need to return the `server_mode`.

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: tests/api2/test_347_posix_mode.py (hunk 3 above)
Comment (anodos325) [reply tests/api2/test_347_posix_mode.py:82]: Sigh. Good eye.

**Review by bmeagherix — APPROVED** [review]

**Pull request merged**